iothub/device/tests/MessageTests.cs

```diff
@@ -165,5 +165,62 @@ public void SettingMessageAsSecurityMessageTest()
             Assert.AreEqual(CommonConstants.SecurityMessageInterfaceId, msg.SystemProperties[MessageSystemPropertyNames.InterfaceId]);
             Assert.IsTrue(msg.IsSecurityMessage);
         }
+
+        [TestMethod]
+        public void CloneWithBodyTest()
+        {
+            using (var original = new Message(Encoding.UTF8.GetBytes("Original copy")))
+            {
+                original.Properties["test1"] = "test_v_1";
+                original.Properties["test2"] = "test_v_2";
+
+                original.ContentEncoding = "gzip";
+                original.ContentType = "text/plain";
+                original.UserId = "JohnDoe";
+
+                using (var clone = original.CloneWithBody(Encoding.UTF8.GetBytes("Cloned version")))
+                { 
+                    Assert.AreEqual("test_v_1", clone.Properties["test1"]);
+                    Assert.AreEqual("test_v_2", clone.Properties["test2"]);
+
+                    Assert.AreEqual("gzip", clone.ContentEncoding);
+                    Assert.AreEqual("text/plain", clone.ContentType);
+                    Assert.AreEqual("JohnDoe", clone.UserId);
+
+                    var clonedContent = default(string);
+
+                    using (var reader = new StreamReader(clone.BodyStream, Encoding.UTF8))
+                    {
+                        clonedContent = reader.ReadToEnd();
+                    }
+                
+                    Assert.AreEqual("Cloned version", clonedContent);
+                }
+            }
+        }
+
+        [TestMethod]
+        public void CloneWithBodyWithNullTest()
+        {
+            using (var original = new Message(Encoding.UTF8.GetBytes("Original copy")))
+            {
+                original.Properties["test1"] = "test_v_1";
+                original.Properties["test2"] = null;
+
+                original.ContentEncoding = "gzip";
+                original.ContentType = null;
+
+                using (var clone = original.CloneWithBody(Encoding.UTF8.GetBytes("Cloned version")))
+                {                    
+                    Assert.AreEqual("test_v_1", clone.Properties["test1"]);
+                    Assert.AreEqual(null, clone.Properties["test2"]);
+                    Assert.AreEqual(2, clone.Properties.Count);
+
+                    Assert.AreEqual("gzip", clone.ContentEncoding);
+                    Assert.AreEqual(null, clone.ContentType);
+                    Assert.IsFalse(clone.SystemProperties.Keys.Contains(MessageSystemPropertyNames.UserId));
+                }
```

Comment:
original only has Properties set, not SystemProperties. What do we want to check with this assertion?

Reply:
There are some system properties opened by getter/setters, e.g. ContentType. In the background those settings are stored in SystemProperties